### DISCUSSION TITLE
Support for Google Maps Plus Codes

### DISCUSSION THREAD

**Discussion by Hund** [organicmaps/organicmaps#1363] [Ideas]
Hi!

It would be nice if there was support for [Google Maps Plus Codes](https://maps.google.com/pluscodes/). I prefer to use Organic Maps, while my partner uses Google Maps. And since OpenStreetMaps is basically empty, I can never search for places we're going to visit. A good workaround would be if she could share the exact location with a Plus Code instead. :)

I know that this is supported by some other client, but I can remember which one though.

**Comment by biodranik** [comment]
Good idea. Is there any open source C++ library for it?

**Reply by Hund** [reply]
If it's of any help, this is the client that I was talking about.

https://github.com/osmandapp/Osmand

I included a screenshot with a search using a Plus Code.

![Screenshot_20210929-215048](https://user-images.githubusercontent.com/320498/135338708-e8a99968-f836-42e0-8317-a92bef8f7ed6.png)

**Reply by Mr-Mime** [reply]
Google released open source libraries for the plus codes.
This one is their c++ library: [https://github.com/google/open-location-code/tree/main/cpp](https://github.com/google/open-location-code/tree/main/cpp)
It is released under the Apache License 2.0, same as organic maps, so this should fit.

I never used the library, but it is what I found when searching for one :D

**Reply by biodranik** [reply]
We're already using it, see 3party dir. Maybe it should be updated though.

**Reply by Mr-Mime** [reply]
Oh never mind.
Just discovered organics maps a few days ago and checked some issues / discussions today and came across this one.
For me it seemed like "still open"

Anyways nice app and thanks for the work, happy to use it on a daily base :)

**Reply by biodranik** [reply]
It is still open, the code doesn't support yet full codes (with cities).
